kotlin/goodmetrics/src/main/kotlin/goodmetrics/downstream/OpentelemetryClient.kt

```diff
@@ -168,7 +174,13 @@ class OpentelemetryClient(
 
     private fun AggregatedBatch.asGoofyOtlpMetricSequence(): Sequence<Metric> = sequence {
         for ((position, measurements) in this@asGoofyOtlpMetricSequence.positions) {
-            val otlpDimensions = position.map { it.asOtlpKeyValue() }
+            // Push down our shared dimensions to each datum leaf if required. For systems that may ingest OTLP metrics
+            // but use a different backing system (e.g. OTLP -> Prometheus)
+            val otlpDimensions = when (prescientDimensions) {
+                is PrescientDimensions.AsDimensions -> position.map { it.asOtlpKeyValue() } + prescientDimensions.sharedDimensions.asOtlpDimensions().toList()
+                is PrescientDimensions.AsMixedResourceDimensions -> position.map { it.asOtlpKeyValue() } + prescientDimensions.resourceAndSharedDimensions.asOtlpDimensions().toList()
+                is PrescientDimensions.AsResource -> position.map { it.asOtlpKeyValue() }
+            }
```

Comment:
```suggestion
            val otlpDimensions = position.map { it.asOtlpKeyValue() } +  when (prescientDimensions) {
                is PrescientDimensions.AsDimensions -> prescientDimensions.sharedDimensions.asOtlpDimensions()
                is PrescientDimensions.AsResource -> emptySequence()
            }
```
I'm not sure I remember why it isn't this way already. What does AsMixedResourceDimensions do that AsDimensions should not not do?

Reply:
Agreed, I believe `AsDimensions` should just work. Fixed in https://github.com/kvc0/goodmetrics_kotlin/pull/35/commits/beb88be149dcaf0a1ddeef7adc20f1b1d69bcdb3